src/utils.rs

```diff
@@ -1,13 +1,37 @@
-use lapin::BasicProperties;
+use lapin::{
+    options::{BasicQosOptions, QueueDeclareOptions},
+    BasicProperties,
+};
 use serde::{Deserialize, Serialize};
 use std::time::Duration;
 
+/// QoS configuration for the AMQP consumer.
+#[derive(Clone, Debug)]
+pub struct QosConfig {
```

Comment:
```suggestion
pub struct QosOptions {
```

Reply:
Lets follow lapin's style and use `Options` instead of `Config` which is repetitive. Repeat for the methods too